src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt

```diff
@@ -82,7 +82,7 @@ fun findImplsAndTraits(project: Project, ty: Ty): Pair<Collection<BoundElement<R
 
     //  XXX: TyStr is TyPrimitive, but we want to handle it separately
         is TyStr -> RsImplIndex.findImpls(project, ty).map { impl -> BoundElement(impl) } to noTraits
-        is TyPrimitive, is TyUnit, is TyUnknown -> noImpls to noTraits
+        is TyUnit, is TyUnknown -> noImpls to noTraits
```

Comment:
I am unsure about this part.
My gut feeling says the following:
We should use the TyStr path for all primitive types.
We should use the else path for TyPointer/TyReference since they need to resolve type parameters.

I also think for TyPointers the type parameter machinery from TyStructOrEnum needs to be adapted/generalized.
Because currently the following code does not resolve types
```rust
let my_ptr: *const char = 42 as *const _;
let new_ptr = my_ptr.offset(42); // evaluates to *const T instead of *const char
```